text/0000-generic-boolean-operators.md

```diff
@@ -0,0 +1,123 @@
+
+## Note on short-circuit operators
+
+Due to the nature of short-circuit operators, it is assumed that `&&` and `||` will not be able to be overloaded in a similar way. If it were, how would it determine whether or not to do the second operation, since it's not required to be a `bool`?
+
+# Drawbacks
+
+## Backward incompatible change
+
+It's a backward-incompatible change, much too close to 1.0 to be comfortable. That's also a reason to do it, because we won't be willing to make backward incompatible changes after 1.0 is released.
+
+## Different naming than other languages
+`Eq` and `Ord` are names that have history in other languages. We don't want to be different without due consideration.
+
+# Alternatives
+
+## Current traits, custom return types
+
+Continue to use the `Eq` and `Ord` traits and their partial forms, but modify them to allow returning types other than `bool`. This is inelegant, and lessens the semantic meaning of implementing `Eq` and `Ord`. It also makes the ergonomics of `Eq` and `Ord` much more unpleasant.
+
+## Allow `Eq` to be implemented in `std::ops`
+
+The naming conflict with `Eq` doesn't necessarily mean that the current `Eq` needs to be renamed to `Equiv`. Since they are in different namespaces, they _could_ co-exist. One of them wouldn't ever be able to be imported in the standard prelude, though.
+
+## Use macros
+
+It's been suggested that using an `ast!` macro (not yet implemented) would be a better approach than allowing the boolean operators to be modified in this way. While macros are a most powerful and awesome tool, I think that they do not eliminate the utility of customizing operators in this way.
+
+They are also not nearly as easily parsable by language tools, especially by text editors, as they will need to be taught about either the whole macro system or each macro in order to properly identify tokens for uses such as syntax highlighting.
+
+## Use methods, not operators
+
+It's also been suggested that these operations should really just be methods, and that instead of wanting `==` to return something other than `bool`, we should just create a new trait, and have the `.eq` method on that trait used, and not bother with the operators at all.
+
+In that scenario, we'd need to do:
+
+```
+MyModel.my_field.eq(42).and(MyModel.my_other_field.gt(17))
+```
+
+instead of
+
+```
+MyModel.my_field == 42 & MyModel.my_other_field > 17
+```
+
+It is my opinion that operators are _significantly_ more readable and parsable than operators in this context, and that the return value not being a boolean is apparent in the context. That being said, if allowing the boolean operators to return non-boolean values was not acceptable, then this would probably be the next best option for my use-case.
```

Comment:
[…] operators are _significantly_ more readable and parsable than **methods** […]